pyccl/tests/test_power.py

```diff
@@ -572,3 +572,21 @@ def test_input_nonlin_raises():
         nonlinear_model='halofit')
     assert 'a:b' in cosmo_input._pk_nl
     assert cosmo_input.has_nonlin_power
+
+
+def test_camb_de_model():
+    """Check that the dark energy model for CAMB has been properly defined."""
+    cosmo = ccl.CosmologyVanillaLCDM(
+        transfer_function='boltzmann_camb',
+        extra_parameters={"camb": {"dark_energy_model": "pf"}})
+    with pytest.raises(ValueError):
+        ccl.linear_matter_power(cosmo, 1, 1)
+
+
+def test_camb_correct_de_model():
+    """Check that w is not less than -1, if the chosen dark energy model for
+    CAMB is fluid."""
+    cosmo = ccl.CosmologyVanillaLCDM(
+        transfer_function='boltzmann_camb', w0=-1, wa=-1)
+    with pytest.raises(ValueError):
+        ccl.linear_matter_power(cosmo, 1, 1)
```

Comment:
Apologies for the delay getting back to this. Maybe just add a
```python
cosmo = ccl.CosmologyVanillaLCDM(
        transfer_function='boltzmann_camb', w0=-1, wa=-1, extra_parameters={"camb": {"dark_energy_model": "ppf"}}
)
ccl.linear_matter_power(cosmo, 1, 1)
```
to check that the PPF version is running? That would directly check for the case of the issue that prompted this (#779).